src/main/fc/fc_msp.c

```diff
@@ -2038,31 +2042,53 @@ static mspResult_e mspCommonProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #if defined(OSD) || defined (USE_OSD_SLAVE)
     case MSP_SET_OSD_CONFIG:
         {
-            const uint8_t addr = sbufReadU8(src);
-            // set all the other settings
-            if ((int8_t)addr == -1) {
+            const uint8_t addr1 = sbufReadU8(src);
+            const uint8_t addr2 = sbufReadU8(src);
```

Comment:
The problem is that it is going to break everything that is not updated to include this check, whereas, if you add it to the end, everything will keep working as usual.